main.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"github.com/tidwall/gjson"
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func validateCurrencyCode(CurrencyCode string) bool {
+	// description: validates currency code
+	result := false
+	// todo: work up a real test
+	if CurrencyCode == "is in list of currencies" {
+		result = true
+	} else {
+		result = false
+	}
+	return result
+}
+
+func getRate(CurrencyFrom string, CurrencyTo string) float64 {
+	httpposturl := fmt.Sprintf("https://query1.finance.yahoo.com/v7/finance/chart/%v%v%v", CurrencyFrom, CurrencyTo, "=x?corsDomain=finance.yahoo.com&range=1d&interval=1d")
+	// todo: get rid of bytes.NewBuffer yet set a request header
+	var jsonData = []byte(`{
+		"b": "l"
+	}`)
+
+	request, error := http.NewRequest("GET", httpposturl, bytes.NewBuffer(jsonData))
+	request.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	response, error := client.Do(request)
+	if error != nil {
+		panic(error)
+	}
+	defer response.Body.Close()
+	if response.StatusCode != 200 {
+		fmt.Println("something was wrong. status code was not 200\n check https://en.wikipedia.org/wiki/ISO_4217 for valid currency codes")
+		os.Exit(response.StatusCode)
+	}
+	responseData, _ := ioutil.ReadAll(response.Body)
+	// digs into the json and pulls out the value we want
+	// todo: use std library to pull this value using structs
+	value := gjson.GetBytes(responseData, "chart.result.0.meta.regularMarketPrice")
```

Comment:
Yeah like `todo` says, you don't need a separate library for this, though maybe it does not hurt either. The `encoding/json` is fine.

Then you can parse the JSON into a struct like:

```
type YahooConversionResponse struct {
  // fields the response has...
}

var response YahooConversionResponse
json.Unmarshal([]byte(responseData), &response)
```

Not tested but something like that.

Reply:
the yahoo response-fields struct was in there in an earlier version but I had a syntax issue. Will try again.